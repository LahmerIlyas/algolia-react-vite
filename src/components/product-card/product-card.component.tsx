import type { ReactNode } from 'react'
import { Component } from 'react'
import { Button } from '../button/button.component'
import Typography from '../typography/typography.component'

interface ProductCardProps {
  id: number
  title: string
  imgUrl: string
  content: string
  price: string
  link: string
}

export class ProductCard extends Component<ProductCardProps> {
  render(): ReactNode {
    return (
      <div className="flex flex-col border-2 rounded-4 border-gray-200 px-4 py-4 lg:basis-[30%] grow shrink lg:max-w-[33%]">
        <img
          src={this.props.imgUrl}
          className="rounded-4 border-none h-xs w-full object-cover object-top"
        />
        <Typography as="h4" variant="s4" className="truncate max-w-[100%]">{this.props.title}</Typography>
        <Typography as="h5" variant="caption" className="mt-1 line-clamp-3 mb-4">{this.props.content}</Typography>
        <div className="flex flex-row justify-between items-center mt-auto">
          <Typography as="h4" variant="s4" dangerouslySetInnerHTML={{ __html: this.props.price }} />
          <Button variant="primary">
            Buy Now
          </Button>
        </div>
      </div>
    )
  }
}
