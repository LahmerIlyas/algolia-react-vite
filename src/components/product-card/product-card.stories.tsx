import React from 'react'
import type { ComponentMeta, ComponentStory } from '@storybook/react'

import { ProductCard } from './product-card.component'

export default {
  title: 'Example/ProductCard',
  component: ProductCard,
} as ComponentMeta<typeof ProductCard >

const Template: ComponentStory<typeof ProductCard> = () => (
  <React.Fragment>
    <ProductCard
      content="This is a product description"
      title="This is a product title"
      imgUrl="https://picsum.photos/200/300"
      link="https://picsum.photos/200/300"
      id={1}
      price="123"
    />
  </React.Fragment>
)

export const ExampleStory = Template.bind({})
ExampleStory.args = {
}
