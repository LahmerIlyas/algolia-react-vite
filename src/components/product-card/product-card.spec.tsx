import { describe, expect, it } from 'vitest'
import renderer from 'react-test-renderer'
import { ProductCard } from './product-card.component'

describe('Given product card element', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(
      <ProductCard
        content="Test content"
        id={1}
        price={'20'}
        title="Test title"
        imgUrl="https://picsum.photos/200/300"
        link="https://picsum.photos/200/300"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
