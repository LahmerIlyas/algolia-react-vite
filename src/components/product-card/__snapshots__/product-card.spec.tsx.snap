// Vitest Snapshot v1

exports[`Given product card element > should renders correctly 1`] = `
<div
  className="flex flex-col border-2 rounded-4 border-gray-200 px-4 py-4 grow shrink md:basis-[50%] md:max-w-[48%] lg:basis-[30%] lg:max-w-[30%]"
>
  <img
    className="rounded-4 border-none h-xs w-full object-cover object-top"
    src="https://picsum.photos/200/300"
  />
  <h4
    as="h4"
    className="text-s4 truncate max-w-[100%]"
  >
    Test title
  </h4>
  <h5
    as="h5"
    className="text-caption mt-1 line-clamp-3 mb-4"
  >
    Test content
  </h5>
  <div
    className="flex flex-row justify-between items-center mt-auto"
  >
    <h4
      as="h4"
      className="text-s4"
      dangerouslySetInnerHTML={
        {
          "__html": "20",
        }
      }
    />
    <button
      className="px-4 py-2 rounded-2 border-2 font-medium bg-green-200"
    >
      Buy Now
    </button>
  </div>
</div>
`;
